src/rdkafka_op.c

```diff
@@ -409,6 +409,33 @@ void rd_kafka_q_op_err (rd_kafka_q_t *rkq, rd_kafka_op_type_t optype,
 }
 
 
+/**
+ * @brief Propagate a topic error event to the application on a specific queue.
+ * \p optype should be RD_KAFKA_OP_ERR for generic errors and
+ * RD_KAFKA_OP_CONSUMER_ERR for consumer errors.
+ */
+ void rd_kafka_q_op_topic_err (rd_kafka_q_t *rkq, rd_kafka_op_type_t optype,
+                               rd_kafka_resp_err_t err, int32_t version,
+                               const char *topic, const char *fmt, ...) {
+	va_list ap;
+	char buf[2048];
+	rd_kafka_op_t *rko;
+
+	va_start(ap, fmt);
+	rd_vsnprintf(buf, sizeof(buf), fmt, ap);
+	va_end(ap);
+
+	rko = rd_kafka_op_new(optype);
+	rko->rko_version = version;
+	rko->rko_err = err;
+	rko->rko_u.err.errstr = rd_strdup(buf);
+        rko->rko_u.err.rkm.rkm_rkmessage.rkt =
+                (rd_kafka_topic_t *)rd_kafka_lwtopic_new(rkq->rkq_rk, topic);
```

Comment:
did you consider any alternatives to the lw topics? could you just provide the topic name instead? what use cases does providing the lw topic object enable that a name doesn't?

pros: nice API (may be worth it if the use cases are compelling enough)
cons: poor encapsulation creating a burden anywhere you might want to handle a lw topic masquerading as a topic.

Reply:
Changing to a string would be a breaking change to the public APIs, and providing a fully-blown rd_kafka_topic_t object to report topic-does-not-exist has problems since librdkafka never lets go of the proper topic objects until the client instance is terminated, and changing that would be a far more intrusive change.

Reply:
as for the cons: that is true, but only the legacy public APIs take a topic_t object, all new ones take a topic name.

Reply:
thanks, sounds good to me.

note: i hadn't got to the point of completely understanding everything, just noting out something that looked possibly off. thanks for bearing with me :-)